src/Core/Managed/Shared/Channel/InMemoryChannel.cs

```diff
@@ -147,12 +164,14 @@ public void Dispose()
 
         private void Dispose(bool disposing)
         {
-            if (disposing)
+            if (disposing && !isDisposed)
             {
                 if (this.transmitter != null)
                 {
                     this.transmitter.Dispose();
                 }
+
```

Comment:
Okay, I did that originally but I had to make transmitter not be readonly and changed my mind, I'll put it back.